### PR TITLE
chore: unify configuration options definition

### PR DESCRIPTION
**Pull request by david-luna**
PR to unify the way configuration options are defined and processed.

NOTE: this is a draft and so its description, do not review yet

### Checklist

- [x] Implement code
- [ ] Add tests
- [ ] Update TypeScript typings
- [ ] Update documentation
- [ ] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Review by david-luna — COMMENTED** [review]
/test tav

**Review by david-luna — COMMENTED** [review]
/test tav

File: lib/config/config.js [hunk 1]
```diff
@@ -23,18 +24,18 @@ const {
...
   MINUS_ONE_EQUAL_INFINITY,
   ARRAY_OPTS,
   KEY_VALUE_OPTS,
-  ENV_TABLE,
-  DEFAULTS,
-  CROSS_AGENT_CONFIG_VAR_NAME,
   URL_OPTS,
+  CROSS_AGENT_CONFIG_VAR_NAME,
+  getDefaulOptions,
+  getEnvionmentOptions,
```
Comment (trentm) [inline lib/config/config.js:37]:
```suggestion
  getEnvironmentOptions,
```

File: lib/config/config.js [hunk 2]
```diff
@@ -23,18 +24,18 @@ const {
...
   BYTES_OPTS,
   MINUS_ONE_EQUAL_INFINITY,
   ARRAY_OPTS,
   KEY_VALUE_OPTS,
-  ENV_TABLE,
-  DEFAULTS,
-  CROSS_AGENT_CONFIG_VAR_NAME,
   URL_OPTS,
+  CROSS_AGENT_CONFIG_VAR_NAME,
+  getDefaulOptions,
```
Comment (trentm) [inline lib/config/config.js:36]:
```suggestion
  getDefaultOptions,
```

File: lib/config/schema.js [hunk 3]
```diff
@@ -6,260 +6,709 @@
...
+  },
+  {
+    name: 'errorOnAbortedRequests',
+    type: 'boolean',
+    defaultValue: false,
+    envVar: 'ELASTIC_APM_ERROR_ON_ABORTED_REQUESTS',
+  },
+  {
+    name: 'exitSpanMinDuration',
+    type: 'durationMiliseconds',
```
Comment (trentm) [inline lib/config/schema.js:192]:
```suggestion
    type: 'durationMilliseconds',
```

and related references.

File: lib/config/schema.js [hunk 4]
```diff
@@ -6,260 +6,709 @@
...
+    envVar: 'ELASTIC_APM_LOGGER',
+  },
+  {
+    name: 'transport',
+    type: 'function',
+    defaultValue: undefined,
+    internal: true,
+  },
+  {
+    name: 'loggingPreambleData',
```
Comment (trentm) [inline lib/config/schema.js:605]: This is a config var?

**Review by trentm — COMMENTED** [review]
LGTM. Just a couple nits. I wasn't sure if you wanted review yet. There is an "XXX" in there that I assume you want to deal with first.

**Comment by david-luna** [comment]
> There is an "XXX" in there that I assume you want to deal with first.

this was about the default value of `errorMessageMaxLength` which made us ralize the docs need a bit of work. I'm going to add it into this PR

File: lib/config/schema.js (hunk 4 above)
Comment (david-luna) [reply lib/config/schema.js:605]: config does the calculation of all this data in the constructor and place it into a property with that name. Then `Agent.start()` access that data to add som extra values like `activationMethod`, `dependencies` and `startTrace`.

On a second though this new schema can keep values from different sources (env, file, start) so the preamble could be generated on demand and exposed as an API. I'm going to remove it

**Review by trentm — APPROVED** [review]
The convention-commit category on the PR title and commit message should be "chore:" or "refactor:" rather than "feat:" right?

Otherwise, LGTM.

**Comment by david-luna** [comment]
Indeed

**Pull request merged**